src/core/Song.cpp

```diff
@@ -687,6 +681,18 @@ void Song::stop()
 	// remove all note-play-handles that are active
 	Engine::mixer()->clear();
 
+	// Moves the control of the models that were processed on the last frame
+	// back to their controllers.
+	if (!m_oldAutomatedValues.isEmpty())
+	{
+		for (auto it = m_oldAutomatedValues.begin(); it != m_oldAutomatedValues.end(); it++)
+		{
+			AutomatableModel * am = it.key();
+			am->setUseControllerValue(true);
+		}
+	}
+	m_oldAutomatedValues.clear();
```

Comment:
This runs on the UI thread, so races with the audio thread. The same goes for the rest of `Song::stop`, though, so I think it should all be wrapped with `Engine::mixer()->requestChangeInModel();` and `Engine::mixer()->doneChangeInModel();` (apart from the quick return condition at the top).